src/main/config/config_streamer.h

```diff
@@ -57,15 +68,22 @@ typedef struct config_streamer_s {
         config_streamer_buffer_align_type_t w;
     } buffer;
     int at;
-    int err;
+    configStreamerResult_e err;
     bool unlocked;
 } config_streamer_t;
 
 void config_streamer_init(config_streamer_t *c);
 
 void config_streamer_start(config_streamer_t *c, uintptr_t base, int size);
-int config_streamer_write(config_streamer_t *c, const uint8_t *p, uint32_t size);
-int config_streamer_flush(config_streamer_t *c);
+configStreamerResult_e config_streamer_write(config_streamer_t *c, const uint8_t *p, uint32_t size);
+configStreamerResult_e config_streamer_flush(config_streamer_t *c);
 
-int config_streamer_finish(config_streamer_t *c);
-int config_streamer_status(config_streamer_t *c);
+configStreamerResult_e config_streamer_finish(config_streamer_t *c);
+configStreamerResult_e config_streamer_status(config_streamer_t *c);
+
+#if defined(CONFIG_IN_FLASH) || defined(CONFIG_IN_FILE)
+void configUnlock(void);
```

Comment:
Is this really public interface? IMO it should be either `config_streamer_unlock()` (called for all cases, possibly implemented as empty), or it should go into separate .h file (possibly config_streamer_impl.h)

Reply:
moved to c file... no need for public - but must be implemented.